### ISSUE TITLE
"detection" output from "predict with caption" is not as predict with classes

### ISSUE THREAD

**Issue by kobic8** [IDEA-Research/Grounded-Segment-Anything#270]
I followed the notebook for using the output of grounding dino: predict with classes, when I changed to the function: predict with cpation, the output of "detection" is not as the same structure of classes prediction. shouldn't the data structure be the same for a general use?

**Comment by rentainhe** [comment]
> I followed the notebook for using the output of grounding dino: predict with classes, when I changed to the function: predict with cpation, the output of "detection" is not as the same structure of classes prediction. shouldn't the data structure be the same for a general use?

Yes, this is because the post-process of GroundingDINO, which will combine words with similar highly confidence, we will refine this issue in GroundingDINO model
